src/poetry/repositories/pool.py

```diff
@@ -1,59 +1,77 @@
 from __future__ import annotations
 
+from collections import OrderedDict
+from dataclasses import dataclass
+from enum import IntEnum
+from enum import auto
 from typing import TYPE_CHECKING
 
+from poetry.repositories.abstract_repository import AbstractRepository
 from poetry.repositories.exceptions import PackageNotFound
-from poetry.repositories.repository import Repository
 
 
 if TYPE_CHECKING:
     from poetry.core.constraints.version import Version
     from poetry.core.packages.dependency import Dependency
     from poetry.core.packages.package import Package
 
+    from poetry.repositories.repository import Repository
 
-class Pool(Repository):
+
+class RepositoryPriority(IntEnum):
```

Comment:
I like the introduction of the dataclass -- with that present, I think this can be named cleaner as `Priority`.